lib/tls.go

```diff
@@ -92,27 +93,37 @@ var tlsVersions = map[uint16]description{
 	tls.VersionTLS12: {"TLS 1.2", "tls_1_2", 2},
 }
 
+// Fill in a human readable name, extracted from the slug
+func explain_cipher(d description) description {
+	kex_cipher := strings.Split(d.Slug, "_WITH_")
+	// The authenticator is the last element, split on _ and reform
+	encrypt_auth := strings.Split(kex_cipher[1], "_")
+	encrypt := strings.Join(encrypt_auth[:len(encrypt_auth)-1], "_")
+	d.Name = fmt.Sprintf("%s key exchange, %s encryption, %s authentication", kex_cipher[0][len("TLS_"):], encrypt, encrypt_auth[len(encrypt_auth)-1])
```

Comment:
Having encryption/authentication split seems like it might be a bit misleading for AEAD cipher modes. I would split this into just kex/cipher tuple maybe.